Feature/splash

JungMinji08 opened the pull request:
스플래시 스크린 임시로 애니메이션 넣어서 작업했어요.
(앱 들어갈 때 기분 좋으라고..ㅎㅎ)

그리고 status bar 색 바꾸는 composable function을 만들었는데 이게 맞나 모르겠네여..
이 부분도 리뷰 한 번 부탁드려요.

JungMinji08 committed:
Add SplashScreen image and animation

- 스플래시 화면에 들어가는 담글담글 로고 이미지 추가 및 레이아웃 구성
- 3초 딜레이 후 홈화면으로 이동
- 임시로 스플래시 애니메이션 추가

JungMinji08 committed:
Add DataStoreRepository and GetIsUserRegisterUseCase

- DataStoreRepository 생성
    - fun getDataStore[T]: Flow<T?>
    - suspend fun getDataStore[T]Once(key: String): T?
    - suspend fun setDataStore[T](key: String, value: T)
- GetIsUserRegisteredUseCase 생성
    - 회원가입 했는지 안했는지 Preference에 저장해둔 값으로 구분해서 만약 회원가입 했으면 홈화면으로, 안했으면 온보딩 화면으로 이동시킴
- SplashViewModel 생성
- hilt 적용

JungMinji08 committed:
Add SetStatusBarColor function

- System UI Library 추가 (made by google..!)
- presentation/common 디렉토리 생성 후 StatusBar 함수 생성
- color: Color, darkIcon: Boolean 매개변수로 넘겨서 status bar 색 설정 가능

JungMinji08 commented:
force push 죄송...

dahyun1226 commented on app/src/main/java/com/mashup/damgledamgle/di/RepositoryModule.kt:
```diff
@@ -0,0 +1,24 @@
+abstract class RepositoryModule {
+    @Singleton
+    @Binds
+    abstract fun provideDataStoreRepository(impl: DataStoreRepositoryImpl): DataStoreRepository
```
개인적으로 binds쓰면 bindDataStoreRepository같은 네이밍을 선호해요
크게 중요하진 않은듯합니다

dahyun1226 commented on app/src/main/java/com/mashup/damgledamgle/di/UseCaseModule.kt:
```diff
@@ -0,0 +1,24 @@
+
+@Module
+@InstallIn(SingletonComponent::class)
+abstract class UseCaseModule {
```
유즈케이스 모듈까지는 없어도 괜찮지 않을까 싶은데 
두 분은 어떻게 생각하는지 궁금하네요

dahyun1226 commented on app/src/main/java/com/mashup/damgledamgle/MainActivity.kt:
```diff
@@ -13,13 +15,20 @@ import dagger.hilt.android.AndroidEntryPoint
 class MainActivity : ComponentActivity() {
 
     private lateinit var navController: NavHostController
+    private val splashViewModel: SplashViewModel by viewModels()
```
이렇게 되면 전부 여기서 파라미터로 넣어줘야할텐데
제가 알기로는 이렇게 말고 Screen에서 주입받도록도 할 수 있을거에요!

JungMinji08 replied:
아악 감사합니다 ㅠㅠ

dahyun1226 commented on domain/src/main/java/com/mashup/damgledamgle/domain/repository/DataStoreRepository.kt:
```diff
@@ -0,0 +1,24 @@
+ *  Copyright © 2022 MashUp All rights reserved.
+ */
+
+interface DataStoreRepository {
```
개인적인 스타일로는 impl 쓸때는 (단일 구현체?) Repo와 impl 같은 파일에 넣는것을 선호합니당

JungMinji08 replied:
이거 멀티모듈에서 가능한가여? 제가 이해를 잘 못 한걸수도..

dahyun1226 replied:
ㅋㅋㅋㅋㅋㅋㅋㅋ아맞다 이번 플젝 멀티모듈이었지 
죄송합니당 무시해도될듯

dahyun1226 commented on domain/src/main/java/com/mashup/damgledamgle/domain/usecase/onboarding/GetIsUserRegisteredUseCaseImpl.kt:
```diff
@@ -0,0 +1,23 @@
+ *  Copyright © 2022 MashUp All rights reserved.
+ */
+
+class GetIsUserRegisteredUseCaseImpl @Inject constructor(
```
유즈케이스는 interface 상속안해도 괜찮지 않을까 싶어요
모듈 없애는 것과 같은 맥락으로

dahyun1226 commented on app/build.gradle:
```diff
@@ -72,6 +72,9 @@ dependencies {
     implementation "androidx.navigation:navigation-compose:$nav_version"
 
+    // System UI Library
+    implementation "com.google.accompanist:accompanist-systemuicontroller:0.17.0"
```
사소하지만 요것도 버전 빼주면 좋을거같아요

dahyun1226 commented on app/src/main/java/com/mashup/damgledamgle/presentation/feature/splash/SplashScreen.kt:
```diff
@@ -1,25 +1,66 @@
-    )
+fun SplashScreen(navController: NavHostController, viewModel: SplashViewModel) {
+    val scale = remember {
+        androidx.compose.animation.core.Animatable(0f)
```
```suggestion
        Animatable(0f)
```

dahyun1226 commented on app/src/main/java/com/mashup/damgledamgle/presentation/feature/splash/SplashScreen.kt:
```diff
@@ -1,25 +1,66 @@
+        androidx.compose.animation.core.Animatable(0f)
+    }
+
     LaunchedEffect(key1 = true) {
```
ㅋㅋㅋㅋㅋ이거 좀 깜찍하네

dahyun1226 commented on data/src/main/java/com/mashup/damgledamgle/repository/datastore/DataStoreRepositoryImpl.kt:
```diff
@@ -0,0 +1,80 @@
+) : DataStoreRepository {
+    private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "DamgleDamgle_DataStore")
+
+    override fun getDataStoreInt(key: String): Flow<Int?> {
```
DataStore의 함수에 dataStore이라는 네이밍이 들어가는게 적절한지는 잘 모르겠어요
개인적으로는 getIntFlow라고 네이밍을 지었을 것 같습니다

dahyun1226 replied:
+ int, boolean등으로 받지 말고 필요한 부분만(isRegistered 등) 노출시키는 방향도 괜찮다고 생각해요

JungMinji08 replied:
int, boolean등으로 받지 말고 필요한 부분만(isRegistered 등) 노출시키는 방향 ==> 
요거도 생각했었는데 뭔가 DataStore Util 느낌으로 편하게 get/set 할 수 있는 방법을 마련해둔 다음에
isRegistered 같은 key값이나 로직은 유즈케이스에서 관리 & repository 통해서 편하게 get/set 하는걸 상상하고 짰어용

...사실 리뷰 듣고보니까 DataStore 사용하는 부분이 별로 없어서 조금 과한거같기도한데 좀 더 작업 진행해보고 수정할게여~ ㅎㅎ

dahyun1226 replied:
좋아요 좋아요~

JungMinji08 committed:
Refactor

- 코드리뷰 반영
- ViewModel MainActivity부터 주입하지 않고, 사용할 곳에서 hiltViewModel()로 생성
- 유즈케이스 interface 상속 제거, usecase hilt module 제거
- DataStoreRepository 함수들 네이밍 변경
- System Ui 라이브러리 버전 dependencies.gradle로 분리
- 기타 코드 정리
